eventkitd/utils/resumablecompressed/reader.go

```diff
@@ -0,0 +1,63 @@
+package resumablecompressed
+
+import (
+	"compress/zlib"
+	"errors"
+	"io"
+
+	"github.com/jtolio/eventkit/eventkitd/utils/delimited"
+)
+
+type Reader struct {
+	r       *delimited.Reader
+	current io.ReadCloser
+}
+
+func NewReader(base io.Reader) *Reader {
+	r := delimited.NewReader(base)
+	return &Reader{
+		r: r,
+	}
+}
+
+type readerFunc func(p []byte) (n int, err error)
+
+func (f readerFunc) Read(p []byte) (n int, err error) { return f(p) }
+
+func (r *Reader) Read(p []byte) (n int, err error) {
+	if r.current == nil {
+		for r.r.Delimited() {
+			r.r.Advance()
+		}
+
+		var read int64
+		current, err := zlib.NewReader(readerFunc(func(p []byte) (n int, err error) {
+			n, err = r.r.Read(p)
+			read += int64(n)
+			return n, err
+		}))
+		if err != nil {
+			if errors.Is(err, io.ErrUnexpectedEOF) && read == 0 {
```

Comment:
As I understand, this would only happen when the underlying reader immediately returns `io.ErrUnexpectedEOF`, why would it even try to read?

Reply:
if zlib.NewReader gets io.EOF after zero bytes, it replaces it with io.ErrUnexpectedEOF. however, it also returns io.ErrUnexpectedEOF if it parses part of a zlib header before getting to EOF. we need to be able to distinguish between an empty stream and a stream with one or more bytes, hence the wrapper.